src/trapfunc.cpp

```diff
@@ -312,28 +344,22 @@ bool trapfunc::caltrops_glass( const tripoint &p, Creature *c, item * )
     if( c == nullptr ) {
         return false;
     }
-    // tiny animals don't trigger caltrops, they can squeeze between them
-    if( c->get_size() == creature_size::tiny ) {
+    // Only a chance to step on the caltrop, dependent on size
+    if( !x_in_y( occupied_tile_fraction( c->get_size() ), 1.0f ) ) {
         return false;
     }
     c->add_msg_player_or_npc( m_bad, _( "You step on a sharp glass caltrop!" ),
                               _( "<npcname> steps on a sharp glass caltrop!" ) );
-    monster *z = dynamic_cast<monster *>( c );
-    if( z != nullptr ) {
-        z->mod_moves( -z->get_speed() * 0.8 );
-        z->deal_damage( nullptr, bodypart_id( "foot_l" ), damage_instance( damage_cut, rng( 9, 15 ) ) );
-        z->deal_damage( nullptr, bodypart_id( "foot_r" ), damage_instance( damage_cut, rng( 9, 15 ) ) );
-    } else {
-        for( const bodypart_id &bp : c->get_ground_contact_bodyparts() ) {
-            c->deal_damage( nullptr, bp, damage_instance( damage_cut, rng( 9, 30 ) ) );
-        }
+    if( c->is_monster() ) {
+        add_msg_if_player_sees( p, _( "The %s steps on a sharp glass caltrop!" ), c->get_name() );
```

Comment:
No other traps affected by your PR, including metal caltrop, gets this line. Is it really needed? If yes, maybe add it to other traps as well, for consistency?

Reply:
Apparently add_msg_player_or_npc and <npcname> handles monsters as an overload, but without proper capitalization

![image](https://github.com/user-attachments/assets/2e07a9b7-f206-4fd8-9b73-9bd221acdfda)

More refactor, more refactor..